lib/src/rules/avoid_init_to_null.dart

```diff
@@ -66,21 +68,30 @@ class AvoidInitToNull extends LintRule implements NodeLintRule {
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    final visitor = _Visitor(this);
+    final visitor = _Visitor(this, context);
     registry.addVariableDeclaration(this, visitor);
     registry.addDefaultFormalParameter(this, visitor);
   }
 }
 
 class _Visitor extends SimpleAstVisitor<void> {
   final LintRule rule;
+  final LinterContext context;
 
-  _Visitor(this.rule);
+  final bool nnbdEnabled;
+  _Visitor(this.rule, this.context)
+      : nnbdEnabled = context.analysisOptions.contextFeatures
```

Comment:
It's possible for NNBD to be enabled in some files and not others.  The right way to check for this is using CompilationUnit.featureSet or LibraryElement.isNonNullableByDefault

Reply:
super thanks!